Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -207,14 +216,263 @@ class DefaultRiskLevels @Inject constructor(
+            appConfig.normalizedTimePerExposureWindowToRiskLevelMapping
+        )
+            ?: throw NormalizedTimePerExposureWindowToRiskLevelMappingMissingException()
+
+        return RiskResult(transmissionRiskLevel, normalizedTime, riskLevel)
+    }
+
+    @SuppressLint("BinaryOperationInTimber")
+    override fun aggregateResults(
+        exposureWindowsAndResult: Map<ExposureWindow, RiskResult>,
+        riskCalculationParameters: RiskCalculationParametersOuterClass.RiskCalculationParameters
+    ): AggregatedRiskResult {
+        val uniqueDates = exposureWindowsAndResult.keys
+            .map { it.dateMillisSinceEpoch }
+            .toSet()
+        Timber.d("uniqueDates: ${TextUtils.join(System.lineSeparator(), uniqueDates)}")
+
+        val exposureHistory = uniqueDates.map {
+            exposureDataMapper(
+                it,
+                exposureWindowsAndResult,
+                riskCalculationParameters
+            )
+        }
+
+        exposureHistory.forEach {
+            Timber.d(
+                "(date=%s," +
+                    " riskLevel=%s," +
+                    " minimumDistinctEncountersWithLowRisk=%s," +
+                    " minimumDistinctEncountersWithHighRisk=%s)",
+                it.date,
+                it.riskLevel,
+                it.minimumDistinctEncountersWithLowRisk,
+                it.minimumDistinctEncountersWithHighRisk
+            )
+        }
+
+        // 6. Determine `Total Risk`
+        val totalRiskLevel =
+            if (exposureHistory.any {
+                    it.riskLevel == RiskCalculationParametersOuterClass
+                        .NormalizedTimeToRiskLevelMapping
+                        .RiskLevel
+                        .HIGH
+                }) {
+                RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.HIGH
+            } else {
+                RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.LOW
+            }
+
+        Timber.d("totalRiskLevel: ${totalRiskLevel.name} (${totalRiskLevel.ordinal})")
+
+        // 7. Determine `Date of Most Recent Date with Low Risk`
+
+        // 8. Determine `Date of Most Recent Date with High Risk`
+
+        // 9. Determine `Total Minimum Distinct Encounters With Low Risk`
+
+        // 10. Determine `Total Minimum Distinct Encounters With High Risk`
+
+        TODO("Implement last steps and adjust values")
+
+        return AggregatedRiskResult(
+            totalRiskLevel,
+            totalMinimumDistinctEncountersWithLowRisk = 1,
+            totalMinimumDistinctEncountersWithHighRisk = 1,
+            mostRecentDateWithLowRisk = 1,
+            mostRecentDateWithHighRisk = 1
+        )
+    }
+
+    private fun exposureDataMapper(
+        date: Long,
+        exposureWindowsAndResult: Map<ExposureWindow, RiskResult>,
+        riskCalculationParameters: RiskCalculationParametersOuterClass.RiskCalculationParameters
+    ): ExposureData {
```

Comment:
`ExposureData` is very generic

Reply:
Yes too generic. I refactored it to AggregatedRiskPerDateResult.
I think its a more clear naming.